src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -235,14 +235,9 @@ public ProgrammingSubmissionResource(ProgrammingSubmissionService programmingSub
         // as the VCS-server performs the request
         SecurityUtils.setAuthorizationObject();
 
-        Exercise exercise = exerciseService.findOne(exerciseId);
-        if (!(exercise instanceof ProgrammingExercise)) {
-            log.warn("REST request to inform about changed test cases of non existing ProgrammingExercise : {}", exerciseId);
-            return ResponseEntity.notFound().build();
-        }
+        programmingExerciseService.setTestCasesChanged(exerciseId, true);
```

Comment:
I guess some documentation would be good here why we set this attribute to true and what implications it has

Reply:
You are right, I added comments to the different uses of the method.